static/app/views/replays/detail/console/consoleMessage.tsx

```diff
@@ -45,33 +47,45 @@ function MessageFormatter({breadcrumb}: MessageFormatterProps) {
     ? sprintf.parse(message).filter(parsed => Array.isArray(parsed))
     : [];
 
-  // TODO `%c` is console specific, it applies colors to messages
-  // for now we are stripping it as this is potentially risky to implement due to xss
-  const consoleColorPlaceholderIndexes = placeholders
-    .filter(([placeholder]) => placeholder === '%c')
-    .map((_, i) => i);
+  // We check if there are placeholders to print the message accordingly with their correct values
+  if (placeholders.length) {
+    // TODO `%c` is console specific, it applies colors to messages
+    // for now we are stripping it as this is potentially risky to implement due to xss
+    const consoleColorPlaceholderIndexes = placeholders
+      .filter(([placeholder]) => placeholder === '%c')
+      .map((_, i) => i);
 
-  // Retrieve message formatting args
-  const messageArgs = args.slice(0, placeholders.length);
+    // Retrieve message formatting args
+    const messageArgs = args.slice(0, placeholders.length);
 
-  // Filter out args that were for %c
-  for (const colorIndex of consoleColorPlaceholderIndexes) {
-    messageArgs.splice(colorIndex, 1);
-  }
+    // Filter out args that were for %c
+    for (const colorIndex of consoleColorPlaceholderIndexes) {
+      messageArgs.splice(colorIndex, 1);
+    }
+
+    // Attempt to stringify the rest of the args
+    const restArgs = args.slice(placeholders.length).map(renderString);
 
-  // Attempt to stringify the rest of the args
-  const restArgs = args.slice(placeholders.length).map(renderString);
+    const formattedMessage = isMessageString
+      ? vsprintf(message.replaceAll('%c', ''), messageArgs)
+      : renderString(message);
 
-  const formattedMessage = isMessageString
-    ? vsprintf(message.replaceAll('%c', ''), messageArgs)
-    : renderString(message);
+    logMessage = [formattedMessage, ...restArgs].join(' ').trim();
+
+    // If there're no placeholders then we should make sure there are no objects that we want to print
+    // as the message prop do not print this objects correctly we need to handle pretty printing them
+  } else if (breadcrumb.message?.includes('[object Object]')) {
+    const argValues = breadcrumb.data?.arguments.map(renderString);
+
+    logMessage = argValues.join(' ').trim();
+  }
 
   // TODO(replays): Add better support for AnnotatedText (e.g. we use message
   // args from breadcrumb.data.arguments and not breadcrumb.message directly)
   return (
     <AnnotatedText
       meta={getMeta(breadcrumb, 'message')}
-      value={[formattedMessage, ...restArgs].join(' ')}
+      value={logMessage || breadcrumb.message}
```

Comment:
I think you probably want to use ?? instead of || here since logMessage is always at least ''

Reply:
disregard it's the end of the day and I had the operators mixed up lol